exe-unit/runtime-api/examples/runtime-server-mock.rs

```diff
@@ -41,10 +56,41 @@ impl server::RuntimeService for RuntimeMock {
     }
 }
 
-impl server::RuntimeEvent for EventMock {}
+async fn runtime_factory<E: RuntimeEvent>(event_emiter: E) -> RuntimeMock<E> {
+    RuntimeMock { event_emiter }
+}
+
+// client
+
+struct EventsData {
+    last_status: ProcessStatus,
+}
+
+struct EventMock(Arc<Mutex<EventsData>>);
```

Comment:
`EventsData` instead of `ProcessStatus` seems to be an extra indirection with no added value

Reply:
I need a type that implements both RuntimeEvent trait and Clone, where each clone shares the same underlying data.

Reply:
done